dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumEventConsumer.kt

```diff
@@ -6,13 +6,157 @@
+            val mappedEvent = map(event, eventType, rumContext, parsedTags)
+            if (mappedEvent != null) {
+                dataWriter.write(mappedEvent)
+            }
+        }
+    }
+
+    private fun map(
+        event: JsonObject,
+        eventType: String,
+        rumContext: RumContext,
+        tags: Map<String, String>
+    ): JsonObject? {
+        return try {
+            when (eventType) {
+                VIEW_EVENT_TYPE -> {
+                    val parsedViewEvent = ViewEvent.fromJson(event.toString())
+                    webRumEventMapper.mapViewEvent(
+                        parsedViewEvent,
+                        rumContext,
+                        getOffset(parsedViewEvent.view.id),
+                        tags
+                    ).toJson().asJsonObject
+                }
+                ACTION_EVENT_TYPE -> {
+                    val parsedActionEvent = ActionEvent.fromJson(event.toString())
+                    webRumEventMapper.mapActionEvent(
+                        parsedActionEvent,
+                        rumContext,
+                        getOffset(parsedActionEvent.view.id),
+                        tags
+                    ).toJson().asJsonObject
+                }
+                RESOURCE_EVENT_TYPE -> {
+                    val parsedResourceEvent = ResourceEvent.fromJson(event.toString())
+                    webRumEventMapper.mapResourceEvent(
+                        parsedResourceEvent,
+                        rumContext,
+                        getOffset(parsedResourceEvent.view.id),
+                        tags
+                    ).toJson().asJsonObject
+                }
+                ERROR_EVENT_TYPE -> {
+                    val parsedErrorEvent = ErrorEvent.fromJson(event.toString())
+                    webRumEventMapper.mapErrorEvent(
+                        parsedErrorEvent,
+                        rumContext,
+                        getOffset(parsedErrorEvent.view.id),
+                        tags
+                    ).toJson().asJsonObject
+                }
+                LONG_TASK_EVENT_TYPE -> {
+                    val parsedLongTaskEvent = LongTaskEvent.fromJson(event.toString())
+                    webRumEventMapper.mapLongTaskEvent(
+                        parsedLongTaskEvent,
+                        rumContext,
+                        getOffset(parsedLongTaskEvent.view.id),
+                        tags
+                    ).toJson().asJsonObject
+                }
+                else -> {
+                    sdkLogger.e(WRONG_EVENT_TYPE_ERROR_MESSAGE.format(US, eventType))
+                    null
+                }
+            }
+        } catch (e: JsonParseException) {
+            sdkLogger.e(JSON_PARSING_ERROR_MESSAGE, e)
+            null
+        }
+    }
+
+    private fun getOffset(viewId: String): Long {
+        // The JS bridge functions are being called from a different Thread so it is very
```

Comment:
Do you mean that multiple JS calls can be dispatched to the native side in different threads ? I'd have thought they would all be processed in the same thread ?

Reply:
Well I am actually not sure about this and this may depend on the WebKit version, I was just thinking just to make sure we handle this just in case. In case there is a chance that this is handled differently on one specific WebKit version then we will have problems.

Reply:
Could you make a few tests to know exactly if this can **really** happen / refer to official documentation if this is mentionned? I'd rather have a solid basis to decide if we do need this or not.

Reply:
yes I did tests on my end, seems that the thread is the same but it looks like a pool thread and I've seen some stackoverflow mentions that WebView is using a thread pool for the bridge. I am going to try to find the source code even though I tried it before and could not find the native implementation in the Webkit where they are doing this. Will keep you posted.